test/tests/functional/pbs_test_tpp.py

```diff
@@ -825,10 +920,162 @@ def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
+                wait_thread = -1
+            elif wait_thread == "T":
+                wait_thread = 4
+            _msg = "No of wait threads is not equal to %s" % wait_thread
+            self.assertEqual(num_wait_thread, wait_thread, _msg)
+            exp_msg = ["pbs_comms should have at least 2 threads",
+                       "tpp init failed"]
+            if wait_thread == 1:
+                for msg in exp_msg:
+                    self.comm.log_match(msg)
+            elif wait_thread == 101:
+                exp_msg[0] = "pbs_comms should have <= 100 threads"
+                for msg in exp_msg:
+                    self.comm.log_match(msg)
+
+    def common_steps_log_events(
+            self, hook_name, existence, starttime=None):
+        """
+        This function contains common steps for test "PBS_COMM_LOG_EVENTS".
+        The steps are to verify the log msgs in comm logs whenever a hook
+        is created, hook script is imported and hook is deleted.
+        :param hook_name: Name of the hook
+        :type hook_name: String.
+        :param existence: If True ,check for existence of
+                        given msg, else check for non-existence of
+                        given msg.
+        :type existence: bool.
+        :param starttime: If set ignore matches that occur before
+                          specified time
+        :type starttime: float
+        """
+        server_ip = socket.gethostbyname(self.server.hostname)
+        attrs = {'event': 'execjob_begin', 'enabled': 'True'}
+        self.server.create_hook(hook_name, attrs)
+        exp_msg = ["MCAST packet from %s:15001" % server_ip,
+                   "mcast done"
+                   ]
+        for msg in exp_msg:
+            self.comm.log_match(msg, existence=existence, starttime=starttime)
+        self.server.import_hook(hook_name, body="import pbs")
+        for msg in exp_msg:
+            self.comm.log_match(msg, existence=existence, starttime=starttime)
+        self.server.manager(MGR_CMD_DELETE, HOOK, id=hook_name)
+        for msg in exp_msg:
+            self.comm.log_match(msg, existence=existence, starttime=starttime)
+
+    def test_PBS_COMM_LOG_EVENTS(self):
+        """
+        Test for verifying the allowable values for PBS_COMM_LOG_EVENTS
+        """
+        a = [0, 511, 2047, "T"]
+        for log_event in a:
+            hook_name = "begin_" + str(log_event)
+            attrib = {'PBS_COMM_LOG_EVENTS': log_event}
+            if log_event == 0 or log_event == "T":
+                start_time = time.time()
+                existence = False
+            else:
+                existence = True
+            self.set_pbs_conf(host_name=self.mom.shortname, conf_param=attrib)
+            self.common_steps_log_events(hook_name, existence, start_time)
+
+    @requirements(num_moms=2, no_mom_on_server=True, num_comms=3)
+    def test_isolated_mom_pools(self):
+        """
+        Test isolated mom pools
+        Configuration:
```

Comment:
One common suggestion I have for all test cases is better to name the hostname variable used for each of these below in docstring. This would make the test cases more readable. common_setup() method has the docstring explanation, but everytime referring that would not be convenient. 
Ex:
Node 1 : Server, Sched, Comm (self.hostA)
        Node 2 : Mom (self.hostB)
        Node 3 : Comm (self.hostD)
        Node 4 : Mom (self.hostC)
        Node 5 : Comm (self.hostE)